edge-util/src/Microsoft.Azure.Devices.Edge.Util/azureLogAnalytics/AzureLogAnalytics.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util.AzureLogAnalytics
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Security.Cryptography;
+    using System.Text;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    /* Sample code from:
+    /* https://github.com/veyalla/MetricsCollector/blob/master/modules/MetricsCollector/AzureLogAnalytics.cs
+    /* https://dejanstojanovic.net/aspnet/2018/february/send-data-to-azure-log-analytics-from-c-code/
+    */
+
+    public sealed class AzureLogAnalytics
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<AzureLogAnalytics>();
+        static AzureLogAnalytics instance = null;
+
+        AzureLogAnalytics()
+        {
+        }
+
+        public static AzureLogAnalytics GetInstance()
+        {
+            Preconditions.CheckNotNull(instance);
+            return instance;
+        }
+
+        public static AzureLogAnalytics InitInstance()
```

Comment:
you don't need this method at all, refer to https://csharpindepth.com/articles/singleton 4th version without static constructor.

Reply:
I like the 3rd method better because it's more explicit of what it is doing.

Reply:
Update the singleton method.